Sources/TSCBasic/Path.swift

```diff
@@ -506,21 +580,25 @@ private struct WindowsPath: Path, Sendable {
     var components: [String] {
         let normalized: UnsafePointer<Int8> = string.fileSystemRepresentation
         defer { normalized.deallocate() }
-
-        return String(cString: normalized).components(separatedBy: "\\").filter { !$0.isEmpty }
+        // Remove prefix from the components, allowing for comparison across normalized paths.
+        return Self.stripPrefix(String(cString: normalized)).components(separatedBy: #"\"#).filter { !$0.isEmpty }
     }
 
     var parentDirectory: Self {
         return self == .root ? self : Self(string: dirname)
     }
 
     init(string: String) {
-        if string.first?.isASCII ?? false, string.first?.isLetter ?? false, string.first?.isLowercase ?? false,
-           string.count > 1, string[string.index(string.startIndex, offsetBy: 1)] == ":"
+        let noPrefixPath = Self.stripPrefix(string)
+        let prefix = string.replacingOccurrences(of: noPrefixPath, with: "") // Just the prefix or empty
+
+        // Perform drive designator normalization i.e. 'c:\' to 'C:\' on string.
```

Comment:
It's concerning that the Win32 APIs don't do this. I wonder if we want to move this part into `canonicalPathRepresentation`.

@compnerd Any thoughts on drive letter casing?

Reply:
If we're canonicalising here so that plain string comparison works, we should use uppercase for device names/drive letters, IMO.

Reply:
Paths are case insensitive, and I don't know of any win32 API that would change the drive letter. I don't know if there is a canonical spelling for the drive letter.